changelog/pending/20250501--programgen-nodejs--add-the-ability-to-return-scalars-from-component-methods.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: programgen/nodejs
+  description: Add the ability to return scalars from component methods (sdk changes)
```

Nit: As your comments later on call out, it's not just scalars, and also methods don't have to be on components, so maybe something like this?
```suggestion
  description: Add `callSingle` to the NodeJS SDK to support non-object method return values
```